### PR TITLE
Add fixer for the `no-only-test` rule

### PR DESCRIPTION
**Pull request by florianb**

### CODE REVIEW

File: rules/no-only-test.js [hunk 1]
```diff
@@ -13,7 +27,12 @@ const create = context => {
 			if (ava.hasTestModifier('only')) {
 				context.report({
 					node,
-					message: '`test.only()` should not be used.'
+					message: '`test.only()` should not be used.',
+					fix: fixer => {
+						const range = getTestModifier(node, 'only').range;
+						range[0] -= 1;
+						return fixer.remove(getTestModifier(node, 'only'));
```
Comment (sindresorhus) [inline rules/no-only-test.js]: I would do:

```js
const range = getTestModifier(node, 'only').range.slice();
range[0] -= 1;
return fixer.removeRange(range);
```

File: rules/no-only-test.js [hunk 2]
```diff
@@ -2,6 +2,20 @@
 const visitIf = require('enhance-visitors').visitIf;
 const createAvaRule = require('../create-ava-rule');
 
+const getTestModifier = function (node, mod) {
```
Comment (sindresorhus) [inline rules/no-only-test.js]: Use arrow-function.

Comment (sindresorhus) [inline rules/no-only-test.js]: And move it to `util.js`

File: rules/no-only-test.js [hunk 3]
```diff
@@ -2,6 +2,20 @@
...
+		return getTestModifier(node.callee, mod);
+	} else if (node.type === 'MemberExpression') {
+		if (node.property.type === 'Identifier' && node.property.name === mod) {
+			return node.property;
+		}
+
+		return getTestModifier(node.object, mod);
+	}
+
+	return undefined;
```
Comment (sindresorhus) [inline rules/no-only-test.js]: This is not needed.

**Comment by sindresorhus** [comment]
Can you also mark it `fixable` in the docs, like done here https://github.com/sindresorhus/eslint-plugin-unicorn#rules

File: rules/no-only-test.js (hunk 2 above)
Comment (florianb) [reply rules/no-only-test.js]: ✔️

File: rules/no-only-test.js (hunk 1 above)
Comment (florianb) [reply rules/no-only-test.js]: ✔️

**Comment by florianb** [comment]
Okay @sindresorhus - i finally made the changes, apologies for the delay.

Moving `getTestModifier` to `util` made it necessary using a named function instead of an arrow function. Let me know if further changes are needed. 🍸

**Comment by jfmengels** [comment]
Hi @florianb, thanks for your work on this :)

I would like to add three cases to the tests. The third one fails and currently creates incorrect JS code (`test.(t => { t.pass(); });`).

```js
		{
			code: header + 'test\n  .only(t => { t.pass(); });',
			output: header + 'test\n  (t => { t.pass(); });',
			errors
		},
		{
			code: header + 'test  .only(t => { t.pass(); });',
			output: header + 'test  (t => { t.pass(); });',
			errors
		},
		{
			code: header + 'test.\n  only(t => { t.pass(); });',
			output: header + 'test\n  (t => { t.pass(); });',
			errors
		}
```

That said, I'm actually against having this rule be fixable :thinking: 
I use autofix a lot now (manually), for things like formatting (especially since I started using prettier and a lot of core rules became autofixable), and some people even have it run automatically on save. Having this rule be fixable means that you can not run the autofix (or even save your file?) while you want to run some tests with `only`.

**Comment by florianb** [comment]
Hi @jfmengels - thanks for suggesting the test-cases, they are perfectly valid!

I absolutely understand your opinion, to be hones, i picked that rule since i wanted to help out and the `no-only`-rule wasn't discussed controversial in #21. I will wait with any further activities until the members found a consent on this.

But to add something to the discussion - i use fixes to make my current work compliant before pushing to the remote.

**Comment by sindresorhus** [comment]
> Having this rule be fixable means that you can not run the autofix (or even save your file?) while you want to run some tests with only.

That can be solved in the editor plugins though, like https://github.com/AtomLinter/linter-eslint/pull/795.

**Comment by jfmengels** [comment]
Ah, I didn't know about that, that's great to hear.
I think most people will not use that, but it's probably worth trying adding it. If it bothers a lot of people, we can always explain in the docs how to disable the autofixing in the editor.
:+1: for me then, pending the addition of the tests that I put above.

**Comment by sindresorhus** [comment]
And in atom-linter-xo, I can disable the auto-fix for it by default.

**Comment by florianb** [comment]
Aye - thanks, going to add the test. ⏳

**Comment by florianb** [comment]
@jfmengels - i added additional tests using tabs instead of spaces.
@sindresorhus - done, even though i guess you'll have to refine my style for your final pleasure. 😄

**Pull request merged**